docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+
+Each <code>Item</code> that contains the keyword is then added to a new <code>ArrayList</code> named
+ <code>filteredItems</code> that is maintained by the <code>FindCommand</code>, which then prints the list to standard output once it is done
+creating the list.
+
+The following sequence diagram below shows how the <code>Shoco</code> object creates the <code>FindCommand</code> object. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/Find.png)
+
+This next sequence diagram will show how the <code>FindCommand</code> creates the <code>filteredItems</code> list:
+
+![alt text](images/FindItem.png)
+
+#### 3.4.2 Design considerations
+
+##### Aspect: Data structure to support the find feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>FindCommand</code>
+ 
+  - Pros: Easy to add the find feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement find feature in the <code>Shoco</code> class
+
+  - Pros: Will have less code to deal with as a new function is simply created in the <code>Shoco</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Shoco</code> class
+    simply executes those commands as black boxes, without worrying about their internal details
+<!-- @@author -->
+  
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+### 3.5 Mark and Unmark feature
+#### 3.5.1 Current Implementation
+  
+ The mark and unmark feature is implemented using the <code>MarkCommand</code> and <code>UnmarkCommand</code> class
+ which extends the main <code>Command</code> class with an index representing that of the item to be marked or
+ unmarked as bought in the list.
+ 
+ The process of object creation:
+ 1. The <code>Duke</code> class first receives user input from the <code>Ui</code>
+ 2. The <code>Duke</code> class then creates a <code>Parser</code> object and calls its <code>parseCommand</code> method
+ to instantiate a <code>MarkCommand</code> or <code>UnmarkCommand</code> object based on the user input
+ 3. The <code>Duke</code> class then calls the <code>execute</code> method of the <code>MarkCommand</code> or 
+ <code>UnmarkCommand</code> command object. This calls the <code>markAsBought</code> or <code>unmarkAsBought</code>
+ method of the <code>shoppingList</code> object with the specified index.
+
+ The following sequence diagram below shows how the Mark feature (Diagram 1) and Unmark feature (Diagram 2) works.
+ Note the <code>Ui</code> class is omitted in the sequence diagram to emphasise on the other classes:
+  
+ Diagram 1:
+ 
+![alt text](images/Mark.png)
+  
+Diagram 2:
+
+![alt text](images/Unmark.png)
```

Comment:
![image](https://user-images.githubusercontent.com/40775612/77994154-5efa7580-72de-11ea-8ec6-b530bf1b4a67.png)

Should the activation bar UnmarkCommand be longer than ShoppingList? I am thinking that control returns to UnmarkCommand after an item is unmarked in ShoppingList